src/index.ts

```diff
@@ -40,78 +29,72 @@ async function bundleMDX(
   }).process(mdx)) as {contents: string}
 
   const dir = path.join(process.cwd(), `__mdx_bundler_fake_dir__`)
-  const entryPath = path.join(dir, './index.mdx.js')
+  const entryPath = path.join(dir, './index.mdx.jsx')
   const absoluteFiles: Record<string, string> = {
     [entryPath]: entryCode,
   }
   for (const [filepath, fileCode] of Object.entries(files)) {
     absoluteFiles[path.join(dir, filepath)] = fileCode
   }
-  const inMemoryModulePlugin = {
-    name: 'in-memory-module',
-    resolveId(importee: string, importer?: string) {
-      if (!importer || importee === entryPath) return importee
-      if (!importee[0].startsWith('.')) return null
-
-      const resolved = path.resolve(path.dirname(importer), importee)
-
-      if (resolved in absoluteFiles) return resolved
-      for (const ext of ['.js', '.tsx', '.jsx', '.ts']) {
-        const resolvedWithExt = `${resolved}${ext}`
-        if (resolvedWithExt in absoluteFiles) return resolvedWithExt
-      }
-      throw new Error(
-        `Could not resolve '${importee}' from ${
-          importer === entryPath
-            ? 'the entry MDX file'
-            : `'${importer.replace(`${dir}/`, '')}'`
-        }`,
-      )
-    },
-    load(id: string) {
-      if (absoluteFiles[id]) return absoluteFiles[id]
-      return null
+
+  const inMemoryPlugin: Plugin = {
+    name: 'inMemory',
+    setup(build) {
+      build.onResolve({filter: /.*/}, args => {
+        if (args.path === entryPath) return {path: args.path}
+
+        const modulePath = path.resolve(path.dirname(args.importer), args.path)
+
+        if (modulePath in absoluteFiles) return {path: modulePath}
+
+        for (const ext of ['.js', '.ts', '.jsx', '.tsx']) {
+          const fullModulePath = `${modulePath}${ext}`
+          if (fullModulePath in absoluteFiles) return {path: fullModulePath}
+        }
+
+        console.dir(args.importer)
```

Comment:
This probably shouldn't still be around.

Reply:
whoops, sorry I'll remove that.